### PR TITLE
Changed PostgreSql New-AzPostgreSqlFlexibleServer cmdlet and Added Get-AzPostgreSqlFlexibleServerLocationBasedCapability

### PR DESCRIPTION
**Pull request by DaeunYim**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@DaeunYim Could you merge https://github.com/Azure/azure-powershell/pull/13552 into this one? I will review them together.

**Comment by DaeunYim** [comment]
> @DaeunYim Could you merge #13552 into this one? I will review them together.

They are already integrated here! And I also resolved the same problems as MySql you would want me to fix here!

**Comment by DaeunYim** [comment]
> @DaeunYim Could you merge #13552 into this one? I will review them together.

They are already integrated here! And I also resolved the same problems as MySql you would want me to fix here!

**Comment by DaeunYim** [comment]
Az.PostgreSql 
Updated New-AzPostgreSqlFlexibleServer cmdlet to provide easy resource group and network management within RDBMS flexible server management. 
Added new cmdlets Get-AzPostgreSqlFlexibleServerLocationBasedCapability and Get-AzPostgreSqlConnectionString

File: src/PostgreSql/custom/Restore-AzPostgreSqlFlexibleServer_PointInTimeRestore.ps1 [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        [Parameter(HelpMessage='The subscription ID that identifies an Azure subscription.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Category('Path')]
+        [Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Runtime.DefaultInfo(Script='(Get-AzContext).Subscription.Id')]
+        [System.String]
+        ${SubscriptionId},
+
+        [Parameter(Mandatory, HelpMessage = 'The point in time to restore from (ISO8601 format), e.g., 2017-04-26T02:10:00+08:00.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Category('Body')]
+        [System.DateTime]
+        ${RestorePointInTime},
```
Comment (msJinLei) [inline src/PostgreSql/custom/Restore-AzPostgreSqlFlexibleServer_PointInTimeRestore.ps1:47]: https://github.com/Azure/azure-powershell/blob/86c11f6258a42ecd0440939ca256ee86e214e576/src/MySql/custom/Restore-AzMySqlFlexibleServer_PointInTimeRestore.ps1#L46

File: src/PostgreSql/custom/Update-AzPostgreSqlFlexibleServer.ps1 [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        # Use the default credentials for the proxy.
+        ${ProxyUseDefaultCredentials}
+    )
+
+    process {
+        try {
+            if ($PSBoundParameters.ContainsKey('AdministratorLoginPassword')) {
+                $bStr = . "$PSScriptRoot/../utils/Unprotect-SecureString.ps1" $PSBoundParameters['AdministratorLoginPassword']
+                $null = $PSBoundParameters.Remove('AdministratorLoginPassword')
+                $PSBoundParameters.Add('AdministratorLoginPassword', [System.Runtime.InteropServices.marshal]::PtrToStringAuto($bStr))
```
Comment (msJinLei) [inline src/PostgreSql/custom/Update-AzPostgreSqlFlexibleServer.ps1:172]: no need to call PtrToStringAuto

File: src/PostgreSql/test/Get-AzPostgreSqlConfiguration.Tests.ps1 [hunk 3]
```diff
@@ -23,7 +23,7 @@ Describe 'Get-AzPostgreSqlConfiguration' {
     }
 
     It 'GetViaIdentity' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/configurations/datestyle"
```
Comment (msJinLei) [inline src/PostgreSql/test/Get-AzPostgreSqlConfiguration.Tests.ps1:26]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Get-AzPostgreSqlFirewallRule.Tests.ps1 [hunk 4]
```diff
@@ -29,7 +29,7 @@ Describe 'Get-AzPostgreSqlFirewallRule' {
 
     It 'GetViaIdentity' {
         New-AzPostgreSqlFirewallRule -Name $env.firewallRuleName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -EndIPAddress 0.0.0.1 -StartIPAddress 0.0.0.0
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/firewallRules/$($env.firewallRuleName)"
```
Comment (msJinLei) [inline src/PostgreSql/test/Get-AzPostgreSqlFirewallRule.Tests.ps1:32]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Get-AzPostgreSqlServer.Tests.ps1 [hunk 5]
```diff
@@ -28,7 +28,7 @@ Describe 'Get-AzPostgreSqlServer' {
     }
 
     It 'GetViaIdentity' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)"
```
Comment (msJinLei) [inline src/PostgreSql/test/Get-AzPostgreSqlServer.Tests.ps1:31]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Get-AzPostgreSqlVirtualNetworkRule.Tests.ps1 [hunk 6]
```diff
@@ -13,25 +13,25 @@ while(-not $mockingPath) {
 
 Describe 'Get-AzPostgreSqlVirtualNetworkRule' {
     It 'List' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/MySqlVNet/subnets/MysqlSubnet1"
```
Comment (msJinLei) [inline src/PostgreSql/test/Get-AzPostgreSqlVirtualNetworkRule.Tests.ps1:16]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Remove-AzPostgreSqlFirewallRule.Tests.ps1 [hunk 7]
```diff
@@ -22,7 +22,7 @@ Describe 'Remove-AzPostgreSqlFirewallRule' {
     It 'DeleteViaIdentity' {
         { 
             New-AzPostgreSqlFirewallRule -Name $env.firewallRuleName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -EndIPAddress 0.0.0.1 -StartIPAddress 0.0.0.0
-            $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/firewallRules/$($env.firewallRuleName)"
```
Comment (msJinLei) [inline src/PostgreSql/test/Remove-AzPostgreSqlFirewallRule.Tests.ps1:25]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Remove-AzPostgreSqlServer.Tests.ps1 [hunk 8]
```diff
@@ -26,7 +26,7 @@ Describe 'Remove-AzPostgreSqlServer' {
             #[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine")]
             $password = 'Pa88word!' | ConvertTo-SecureString -AsPlainText -Force
             New-AzPostgreSqlServer -Name postgresqldelete -ResourceGroupName $env.resourceGroup -Location $env.location -AdministratorUserName pwsh -AdministratorLoginPassword $password -Sku $env.Sku
-            $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/postgresqldelete"
```
Comment (msJinLei) [inline src/PostgreSql/test/Remove-AzPostgreSqlServer.Tests.ps1:29]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Remove-AzPostgreSqlVirtualNetworkRule.Tests.ps1 [hunk 9]
```diff
@@ -13,17 +13,17 @@ while(-not $mockingPath) {
 
 Describe 'Remove-AzPostgreSqlVirtualNetworkRule' {
     It 'Delete' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/MySqlVNet/subnets/MysqlSubnet1"
```
Comment (msJinLei) [inline src/PostgreSql/test/Remove-AzPostgreSqlVirtualNetworkRule.Tests.ps1:16]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Restart-AzPostgreSqlServer.Tests.ps1 [hunk 10]
```diff
@@ -20,7 +20,7 @@ Describe 'Restart-AzPostgreSqlServer' {
 
     It 'RestartViaIdentity' {
         { 
-            $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/restart"
```
Comment (msJinLei) [inline src/PostgreSql/test/Restart-AzPostgreSqlServer.Tests.ps1:23]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Update-AzPostgreSqlConfiguration.Tests.ps1 [hunk 11]
```diff
@@ -18,7 +18,7 @@ Describe 'Update-AzPostgreSqlConfiguration' {
     }
 
     It 'UpdateViaIdentityExpanded' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/configurations/deadlock_timeout"
```
Comment (msJinLei) [inline src/PostgreSql/test/Update-AzPostgreSqlConfiguration.Tests.ps1:21]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Update-AzPostgreSqlFirewallRule.Tests.ps1 [hunk 12]
```diff
@@ -22,7 +22,7 @@ Describe 'Update-AzPostgreSqlFirewallRule' {
 
     It 'UpdateViaIdentityExpanded' {
         New-AzPostgreSqlFirewallRule -Name $env.firewallRuleName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -EndIPAddress 0.0.0.1 -StartIPAddress 0.0.0.0
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/firewallRules/$($env.firewallRuleName)"
```
Comment (msJinLei) [inline src/PostgreSql/test/Update-AzPostgreSqlFirewallRule.Tests.ps1:25]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/test/Update-AzPostgreSqlVirtualNetworkRule.Tests.ps1 [hunk 13]
```diff
@@ -13,19 +13,19 @@ while(-not $mockingPath) {
...
         $rule.VirtualNetworkSubnetId | Should -Be $ID2
         Remove-AzPostgreSqlVirtualNetworkRule -Name $env.VNetName -ResourceGroupName $env.resourceGroup-ServerName $env.serverName
     }
 
     It 'UpdateViaIdentityExpanded' {
-        $SubnetID1 = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/MySqlVNet/subnets/MysqlSubnet1"
+        $SubnetID1 = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/PostgreSqlVnet/subnets/MysqlSubnet1"
         New-AzPostgreSqlVirtualNetworkRule -Name $env.VNetName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -SubnetId $SubnetID1
-        $SubnetID2 = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/MySqlVNet/subnets/MysqlSubnet2"
-        $VNetId = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/virtualNetworkRules/$($env.VNetName)"
```
Comment (msJinLei) [inline src/PostgreSql/test/Update-AzPostgreSqlVirtualNetworkRule.Tests.ps1:28]: Revert manual change of the id. You can skip the similar case currently.

File: src/PostgreSql/custom/New-AzPostgreSqlFlexibleServer.ps1 [hunk 14]
```diff
@@ -0,0 +1,596 @@
...
+            Write-Host $Msg
+            $Server = Az.PostgreSql.internal\New-AzPostgreSqlFlexibleServer @PSBoundParameters
+
+            # Create Firewallrules
+            $FirewallRuleName = CreateFirewallRule $NetworkParameters
+
+            if (![string]::IsNullOrEmpty($FirewallRuleName)) {
+                $Server.FirewallRuleName = $FirewallRuleName
+            }
+            $Server.SecuredPassword =  $PSBoundParameters.AdministratorLoginPassword | ConvertTo-SecureString -AsPlainText -Force
```
Comment (msJinLei) [inline src/PostgreSql/custom/New-AzPostgreSqlFlexibleServer.ps1:307]: No Database name needed here?

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/PostgreSql/docs/Restore-AzPostgreSqlFlexibleServer.md [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+
+# Restore-AzPostgreSqlFlexibleServer
+
+## SYNOPSIS
+Restore a PostgreSQL flexible server from an existing backup
+
+## SYNTAX
+
+```
+Restore-AzPostgreSqlFlexibleServer -Name <String> -ResourceGroupName <String> -SourceServerName <String>
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/PostgreSql/docs/Restore-AzPostgreSqlFlexibleServer.md:16]: Yes, this  is from backend difference.. PG only accepts server name while MySQL also accepts server Id.

File: src/PostgreSql/custom/Update-AzPostgreSqlFlexibleServer.ps1 (hunk 2 above)
Comment (DaeunYim) [reply src/PostgreSql/custom/Update-AzPostgreSqlFlexibleServer.ps1:172]: Removed, thanks.

File: src/PostgreSql/custom/New-AzPostgreSqlFlexibleServer.ps1 (hunk 14 above)
Comment (DaeunYim) [reply src/PostgreSql/custom/New-AzPostgreSqlFlexibleServer.ps1:307]: No database CRUD for PG now. It;s only recently implemented, that it is not included in the current swagger file

File: src/PostgreSql/custom/Restore-AzPostgreSqlFlexibleServer_PointInTimeRestore.ps1 (hunk 1 above)
Comment (DaeunYim) [reply src/PostgreSql/custom/Restore-AzPostgreSqlFlexibleServer_PointInTimeRestore.ps1:47]: Any concern here??

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please generate use  autorest.cmd --use:@autorest/powershell@3.0.414. We have just upgraded the autorest.powershell

File: src/PostgreSql/utils/adjectives.txt [hunk 16]
```diff
@@ -0,0 +1,166 @@
+aboard
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/PostgreSql/utils/adjectives.txt:1]: Will do!

**Comment by DaeunYim** [comment]
> Please generate use autorest.cmd --use:@autorest/powershell@3.0.414. We have just upgraded the autorest.powershell

Rebuilt and pushed a change with the txt files

**Review by msJinLei — APPROVED** [review]

**Pull request merged**